src/test/mockOutputChannel.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+export class MockOutputChannel implements vscode.OutputChannel {
+  public value: string = ''
+  public isHidden: boolean = false
+  public preserveFocus: boolean = false
+
+  public readonly name = 'Mock channel'
+
+  public append(value: string): void {
+    this.value += value
+  }
+
+  public appendLine(value: string) {
+    this.value += value + '\n'
+  }
+
+  public clear(): void {
+    this.value = ''
+  }
+
+  public dispose(): void {
+    this.value = ''
+  }
+
+  public hide(): void {
+    this.isHidden = true
+  }
+
+  /**
+   * This is overloaded by VS Code but viewColumn version is deprecated thus
+   * show(preserveFocus?: boolean) is really what we should consider.
+   */
+  public show(columnOrPreserveFocus?: vscode.ViewColumn | boolean, preserveFocus?: boolean): void {
```

Comment:
Should we throw on `preserveFocus !== undefined`?

Reply:
I don't think so. It is optional. My understanding is that regardless of the the arguments, calling `show` will un-hide the output channel and that preserving focus is tangential which is why all params are optional in the `OutputChannel` interface.